lib/VMwareWebService/broker_timeout.rb

```diff
@@ -0,0 +1,64 @@
+module Timeout
```

Comment:
I think our disagreement is over whether it is in fact a critical section. In my understanding of ruby's execution model, the "duelling threads" scenario can resolve with x copping the raise; y getting killed; or both... and all three of those seem fine, in terms of timeout's expected behaviour. I consider any other result a violation of the VM's contract.